### PR TITLE
Generate production build on npm run build

### PR DESCRIPTION
**Pull request by AnshulMalik**
Associated Issue: #52 

### Summary of Changes

* `npm run build` will now generate a production build
* The size of `main.bundle.js` is checked in the tests

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -66,6 +67,7 @@
     "@types/react": "^16.0.28",
     "@types/react-dom": "^16.0.3",
     "awesome-typescript-loader": "^3.4.1",
+    "chalk": "^2.3.0",
```
Comment (yurydelendik) [inline package.json]: let's remove this dependency

File: bin/check-bundle-size.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env node
+
```
Comment (yurydelendik) [inline bin/check-bundle-size.js]: We need some base license test here, or where it came from.

**Comment by mbebenita** [comment]
@AnshulMalik how do I generate a debug build? For development, I use `npm run build-watch`, will this produce a debug build?

**Comment by AnshulMalik** [comment]
Yep @mbebenita , `build-watch` will produce the debug build.

File: tsconfig.json [hunk 3]
```diff
@@ -6,7 +6,7 @@
       "skipLibCheck": true,
       "module": "esnext",
       "moduleResolution": "node",
-      "target": "esnext",
+      "target": "es5",
```
Comment (yurydelendik) [inline tsconfig.json]: If we can, let's keep it `es2017`. If uglify-js does not play well, can we use uglify-es or some other minifier? If it's not possible, we probably need to keep code non-production and fix limit to 600k.

File: bin/check-bundle-size.js (hunk 2 above)
Comment (AnshulMalik) [reply bin/check-bundle-size.js]: I borrowed this file file from `debugger.html` so what should I write as license?

Comment (yurydelendik) [reply bin/check-bundle-size.js]: you can just write where you borrowed the file, since we are not clear about its license.

**Comment by yurydelendik** [comment]
thank you for the patch

**Pull request merged**